### PR TITLE
BAU: Added initial developer documentation

### PR DESCRIPTION
**Pull request by chris-dco**
### What changed

Updated README.md to provide addition documentation often required by new developers to IPV Core.

### Why did it change

README.md was missing basic documentation to enable a developer to start working on IPV Core.

### Issue tracking

N/A

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed
- [x] Documented in the [README](./blob/main/README.md)

### Other considerations

- [x] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

**Review by kerrr — DISMISSED** [review]
This is much better so happy to tick. We will need to come back and revise again in the context of the dev environment work when that is properly available and documented.

File: README.md [hunk 1]
```diff
@@ -2,127 +2,93 @@
...
-
-
-
-## Pre-Commit Checking / Verification
-
-Completely optional, there is a `.pre-commit-config.yaml` configuration setup in this repo, this uses [pre-commit](https://pre-commit.com/) to verify your commit before actually commiting, it runs the following checks:
-
+The following projects are related to the di-ipv-core-back, providing additional functionality:
+* [di-ipv-core-front](https://github.com/alphagov/di-ipv-core-front) - Front end code for the core of the Identity Proofing and Verification (IPV) system.
+* [di-ipv-core-test](https://github.com/alphagov/di-ipv-core-test) - Feature tests for the core of the Identity Proofing and Verification (IPV) system.
```
Comment (blakeyp) [inline README.md]: It's di-ipv-core-tests

File: README.md [hunk 2]
```diff
@@ -2,127 +2,93 @@
...
+brew install --cask intellij-idea
+brew install --cask docker
+brew install jq
+brew install alphagov/gds/gds-cli
+brew tap aws/tap
+brew install awscli
+brew install aws-sam-cli
+```
+
+We us [pre-commit](https://pre-commit.com/) to help with linting. This configured through the [.pre-commit-config.yaml](pre-commit-config.yaml) configuration setup in this repo, this uses pre-commit to verify your commit before actually commiting, it runs the following checks:
```
Comment (blakeyp) [inline README.md]: *use

File: README.md [hunk 3]
```diff
@@ -2,127 +2,93 @@
...
-```
+### Deployment
+See the [Deployment Documentation](deploy/README.md) for information on deploying the di-ipv-core-back project.
 
-# SAM (Serverless Application Model)
+### Testing
+The di-ipv-core-back has a number of different tests:
+* Unit Tests - Each lambda contains unit tests which test a classes functionality in isolation. These tests can be found within the `lambda\*\src\test` folder.
+* Integration Tests - Integration tests are found in the `integration-test` folder and test the functionality of `di-ipv-core-back` running on an AWS test environment.
+* Feature Tests - Cucumber feature tests for the core of the Identity Proofing and Verification (IPV) system reside in the [di-ipv-core-test](https://github.com/alphagov/di-ipv-core-test) project. The tests run against a deployment of di-ipv-core-back and di-ipv-core-front and test the IPV Core user journeys.
```
Comment (blakeyp) [inline README.md]: di-ipv-core-tests

Comment (chris-dco) [reply README.md]: Done.

File: README.md (hunk 1 above)
Comment (chris-dco) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (chris-dco) [reply README.md]: Done

**Review by chris-dco — COMMENTED** [review]
BAU: Updated spelling mistakes found during review.

**Review by blakeyp — APPROVED** [review]

**Pull request merged**